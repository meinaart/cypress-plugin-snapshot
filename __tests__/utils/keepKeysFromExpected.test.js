
describe('utils/keepKeysFromExpected', () => {
  const expected = {
    field: "value",
    items: [
      {name: "item1", "items": [{name: "subitem1"}]},
      {name: "item2"},
      {name: "item3"},
    ]
  };

  const actual = {
    field: "value",
    extra: "extra",
    items: [
      {name: "item1", "items": [{name: "subitem1", extra: 'extra1'}, {name: "subitem2", extra: 'extra2'}]},
      {name: "item2"},
      {name: "item3"},
      {name: "item4"},
    ]
  };

  it('ignoreExtraFields: false, ignoreExtraArrayItems: false', () => {
    const config = {
      ignoreExtraFields: false,
      ignoreExtraArrayItems: false,
    };
    const keepKeysFromExpected = require('../../src/utils/keepKeysFromExpected');
    const result = keepKeysFromExpected(actual, expected, config);
    expect(result).toMatchSnapshot();
  });

  it('ignoreExtraFields: true, ignoreExtraArrayItems: false', () => {
    const config = {
      ignoreExtraFields: true,
      ignoreExtraArrayItems: false,
    };
    const keepKeysFromExpected = require('../../src/utils/keepKeysFromExpected');
    const result = keepKeysFromExpected(actual, expected, config);
    expect(result).toMatchSnapshot();
  });

  it('ignoreExtraFields: true, ignoreExtraArrayItems: true', () => {
    const config = {
      ignoreExtraFields: true,
      ignoreExtraArrayItems: true,
    };
    const keepKeysFromExpected = require('../../src/utils/keepKeysFromExpected');
    const result = keepKeysFromExpected(actual, expected, config);
    expect(result).toMatchSnapshot();
  });

  it('ignoreExtraFields: false, ignoreExtraArrayItems: true', () => {
    const config = {
      ignoreExtraFields: false,
      ignoreExtraArrayItems: true,
    };
    const keepKeysFromExpected = require('../../src/utils/keepKeysFromExpected');
    const result = keepKeysFromExpected(actual, expected, config);
    expect(result).toMatchSnapshot();
  });
});
