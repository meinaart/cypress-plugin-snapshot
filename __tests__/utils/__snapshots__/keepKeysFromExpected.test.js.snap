// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/keepKeysFromExpected ignoreExtraFields: false, ignoreExtraArrayItems: false 1`] = `
Object {
  "extra": "extra",
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "extra": "extra1",
          "name": "subitem1",
        },
        Object {
          "extra": "extra2",
          "name": "subitem2",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": "item4",
    },
  ],
}
`;

exports[`utils/keepKeysFromExpected ignoreExtraFields: false, ignoreExtraArrayItems: true 1`] = `
Object {
  "extra": "extra",
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "extra": "extra1",
          "name": "subitem1",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
  ],
}
`;

exports[`utils/keepKeysFromExpected ignoreExtraFields: true, ignoreExtraArrayItems: false 1`] = `
Object {
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "name": "subitem1",
        },
        Object {
          "extra": "extra2",
          "name": "subitem2",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": "item4",
    },
  ],
}
`;

exports[`utils/keepKeysFromExpected ignoreExtraFields: true, ignoreExtraArrayItems: true 1`] = `
Object {
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "name": "subitem1",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
  ],
}
`;
